routing/turns.cpp

```diff
@@ -52,6 +52,20 @@ static_assert(g_turnNames.size() == static_cast<size_t>(TurnDirection::Count),
 namespace routing
 {
 // UniNodeId -------------------------------------------------------------------
+std::atomic<uint64_t> UniNodeId::m_nextFakeId(0);
+
+UniNodeId::UniNodeId(FeatureID const & featureId, uint32_t startSegId, uint32_t endSegId,
+                     bool forward)
+  : m_type(Type::Mwm)
+  , m_featureId(featureId)
+  , m_startSegId(startSegId)
+  , m_endSegId(endSegId)
+  , m_forward(forward)
+{
+  if (!m_featureId.IsValid())
+    m_nodeId = static_cast<NodeID>(m_nextFakeId++);
```

Comment:
On the one hand```static const NodeID SPECIAL_NODEID = std::numeric_limits<unsigned>::max();```

On the other hand in mwm case |m_nodeId| should be a relatively small number. In mwm/index_graph case  (when |m_type| == Type::Mwm) |m_nodeId| is equal or less number of fake features near start and finish.

Anyway I'm adding a check:
```
  if (!m_featureId.IsValid())
  {
    m_nodeId = static_cast<NodeID>(m_nextFakeId++);
    CHECK_NOT_EQUAL(m_nodeId, SPECIAL_NODEID, ());
  }
```